### ISSUE TITLE
Notifications not always working

### ISSUE THREAD

**Issue by 3arn0wl** [vector-im/riot-android#3062]
Description

Notifications in the Riot Android app are spasmodic. This includes all messages - i.e. ones which incorporate the users' name, as well as those which don't.
Steps to reproduce

There does not seem to be any reasons why notifications sometimes work, and other times don't.

Describe how what happens differs from what you expected.
Sometimes notifications are given : at other times they are not.

Log: sent/not sent? Sorry - they are on other people's devices.
Version information

The latest version of the Android app, and previous versions too.

    Platform: Android

**Comment by BloodyIron** [comment]
Yeah, I've found recent releases of Riot Android are unreliable in terms of notifying the user they have notifications/messages. Whether it's a direct mention, an at room notification, or even just a regular message in a channel/chat they have set to get notifications for.

This was reliable several weeks ago, but very recently seems inconsistent, and not sure why.

This is also happening for generally anyone else I know using Riot Android

Version 0.8.25 G-b167

**Comment by dkasak** [comment]
Notifications were not working for me and a friend at all on 0.8.29 until I disabled battery optimizations. I'm pretty sure it used to work out of the box and stopped working recently. I'm on Android 9 and my friend is on Android 7.

**Comment by BloodyIron** [comment]
@dkasak would love to hear if that causes tangible power usage increase. In our testing, Riot seems to be very efficient for power usage, but if that's the "solution" (and honestly everyone should not have to worry about whether that's toggled on or not), then I wonder if it causes any adverse results.

I'm of the position that we shouldn't have to be toggling settings for notifications to work reliably. The built-in troubleshooting tool I think is a a fair thing to have, but that's not what I'm getting at. It defeats the point of a tool for communications, if your settings mean you get less notifications than those you're trying to keep touch with.

**Comment by dkasak** [comment]
I'll report if I notice any difference in power consumption.

I agree with your sentiment completely. However, I really think this is a regression since I remember it working and there was some recent work on notifications, IIRC.

For what it's worth, the built-in troubleshooting tool reports everything as okay for both me and my friend.

**Comment by BloodyIron** [comment]
@dkasak yeah I've been seeing some odd regressions too with some of my friends reporting inconsistent rates of notifications. Like, sometimes they won't receive my notifications till way later for unknown reasons. Not sure what changes would have caused that, hmm...

**Comment by echto** [comment]
It might be the latest version of synapse and not the client.  Notifications stopped working with a recently updated HS.  Once the synapse version was rolled back notifications resumed.

**Comment by dkasak** [comment]
No, it was definitely the client for me. Once I disabled battery optimizations, they started getting delivered.

As promised, I'm also reporting that I haven't noticed changes in power consumption with it turned on.

**Comment by BloodyIron** [comment]
@echto and @dkasak might be experiencing different things.
